### PR TITLE
:tada: Amazon Ads - new streams for bids and keyword recommendations

### LINKED ISSUE #26400: Source Amazon Ads: new stream Suggested keywords
### What area the feature impact?

None

### Revelant Information

https://advertising.amazon.com/API/docs/en-us/sponsored-products/keywords/suggestions

### LINKED ISSUE #26402: Source Amazon Ads: new stream Bid recommendations
### What area the feature impact?

Connectors

### Revelant Information

https://advertising.amazon.com/API/docs/en-us/sponsored-display/features#bid-recommendations

### PR DESCRIPTION
**Pull request by midavadim**
## What
**SponsoredProductAdGroupSuggestedKeywords**:
  Latest API:
      https://advertising.amazon.com/API/docs/en-us/sponsored-products/3-0/openapi/prod#/Keyword%20Targets/getRankedKeywordRecommendation
      POST /sp/targets/keywords/recommendations
      **Note: does not work, always get "403 Forbidden"**

  V2 API:
      https://advertising.amazon.com/API/docs/en-us/sponsored-products/2-0/openapi#/Suggested%20keywords
      GET /v2/sp/adGroups/{{adGroupId}}>/suggested/keywords

**SponsoredProductAdGroupBidRecommendations**:
    Latest API:
      https://advertising.amazon.com/API/docs/en-us/sponsored-display/3-0/openapi#/Bid%20Recommendations/getTargetBidRecommendations
      POST /sd/targets/bid/recommendations
      **Note: does not work, always get "403 Forbidden"**

  V2 API:
      https://advertising.amazon.com/API/docs/en-us/sponsored-products/2-0/openapi#/Bid%20recommendations/getAdGroupBidRecommendations
      GET /v2/sp/adGroups/{adGroupId}/bidRecommendations

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
No

### CODE REVIEW

**Review by lazebnyi — COMMENTED** [review]
Can not find version bump for metadata and docker.

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`9a7e9b61a2`](https://github.com/airbytehq/airbyte/pull/28002/commits/9a7e9b61a2a1b1509e9a3cae58391ea0585b2adb)) - ✅

⏲️  Total pipeline duration: 03mn41s 

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check                                           | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/midavadim_26400_26402_amazon_adgroup_bids_keyword_recommendations/1688654712/9a7e9b61a2a1b1509e9a3cae58391ea0585b2adb/source-amazon-ads/2.2.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Comment by midavadim** [comment]
> Can not find version bump for metadata and docker.

updated again, it disappeared after merge

File: airbyte-integrations/connectors/source-amazon-ads/Dockerfile [hunk 1]
```diff
@@ -11,8 +11,8 @@ RUN pip install .
 
 ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
 ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
+d
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-amazon-ads/Dockerfile:14]: typo

**Review by lazebnyi — CHANGES_REQUESTED** [review]

**Review by lazebnyi — APPROVED** [review]

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`10b52eb9f7`](https://github.com/airbytehq/airbyte/pull/28002/commits/10b52eb9f7350fdcdc7522d8e718012cb5f8aa01)) - ✅

⏲️  Total pipeline duration: 03mn32s 

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check                                           | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/midavadim_26400_26402_amazon_adgroup_bids_keyword_recommendations/1688663645/10b52eb9f7350fdcdc7522d8e718012cb5f8aa01/source-amazon-ads/2.3.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Comment by octavia-squidington-iii** [comment]
##  source-amazon-ads test report (commit [`d178fbaa41`](https://github.com/airbytehq/airbyte/pull/28002/commits/d178fbaa410d9b31910aacd738491cb61ef44174)) - ✅

⏲️  Total pipeline duration: 03mn05s 

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-amazon-ads/metadata.yaml | ✅        |
| Connector version semver check                                           | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-amazon-ads docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Acceptance tests                                                         | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/midavadim_26400_26402_amazon_adgroup_bids_keyword_recommendations/1688668944/d178fbaa410d9b31910aacd738491cb61ef44174/source-amazon-ads/2.3.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-amazon-ads test
```

**Pull request merged**